src/runtime/opengl.cpp

```diff
@@ -735,6 +761,26 @@ WEAK bool get_texture_format(void *user_context, buffer_t *buf,
     return true;
 }
 
+// This function returns the width, height and number of color channels that the
+// texture for the specified buffer_t will contain. It provides a single place
+// to implement the logic snapping zero sized dimensions to one element.
+WEAK bool get_texture_dimensions(void *user_context, buffer_t *buf, GLint &width,
+                            GLint &height, GLint &channels) {
+
```

Comment:
I think Halide (also) tends to avoid the non-const-ref-for-out-args pattern (in favor of pointers).